### ISSUE TITLE
Introduce factory for Analytics 4 response mock data

### ISSUE THREAD

**Issue by aaemnnosttv** [google/site-kit-wp#6174]
## Feature Description

A `data-mock` set of utilities should be created for generating and receiving mock report data for use by the `analytics-4` data store, similar to what has been done for `analytics`, `search-console`, and `adsense`.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A `analytics-4/util/data-mock.js` module should be created, similar to the utilities by the same name for other modules
* It should export the following functions:
    * `getAnalytics4MockResponse( args )` – a pure function that generates a mock report in the same format that would be returned from the server backend for a report request with the given `args` 
    * `provideAnalytics4MockReport( registry, options )` – a pure function that generates and receives the resulting mock report data onto the `analytics-4` store of the given registry 

## Implementation Brief

- Create a new file, `assets/js/modules/analytics-4/util/data-mock.js`.
- Copy everything from `assets/js/modules/analytics/util/data-mock.js` to `assets/js/modules/analytics-4/util/data-mock.js`.
- Rename the function `provideAnalyticsMockReport` to `provideAnalytics4MockReport`.
- Rename the function `getAnalyticsMockResponse` to `getAnalytics4MockResponse`.
- Replace all instances of `MODULES_ANALYTICS` with `MODULES_ANALYTICS_4`.
- Replace all instances of `/analytics/i` with `/analytics-4/i`.
- Refer to the metric & dimension mappings in the [Phase 2 docs](https://docs.google.com/document/d/1X2TM_kZLZ-LZP4KdyNITQTdd9Ts1J2DQuNwoAEyT-3s/edit?usp=sharing&resourcekey=0-3n4FbMw2uSMKVEkOmbc_jQ) and the [migration guide](https://developers.google.com/analytics/devguides/migration/api/reporting-ua-to-ga4-dims-mets), update the mock metrics/dimensions as appropriate and remove those which are no longer relevant (e.g., the AdSense ones as GA4 isn't yet linked with AdSense).

### Test Coverage

- Create a new file, `assets/js/modules/analytics-4/util/data-mock.test.js`.
- Add test coverage for the `getAnalytics4MockResponse` function.

## QA Brief
* At this point, the changes in the PR does not affect the plugin since the functions are not called anywhere, so no regular QA needed here.
* Because of this, QA can be skipped for this issue and it moved directly to the Approval queue.

## Changelog entry

* N/A.

**Comment by aaemnnosttv** [comment]
@FlicHollis As a note about the dependency here, it's more specific to one part of it (the `provide*` function). The majority of the work here is around generating the mock data and would not be blocked by the selector issue. Also worth noting that @eugene-manuilov has worked on most if not all of these in the past so it might be worth at least having him write or approve the IB here.

**Comment by FlicHollis** [comment]
Thanks so much for the additional context @aaemnnosttv

**Comment by techanvil** [comment]
Hey @hussain-t, the initial approach is looking good in the IB, but I think it would benefit from going into a little more detail on how to achieve the last point.

> Ensure the mock response matches the format of the response from the Analytics 4 getReport selector.

For example, the advice might be to refer to the metric & dimension mappings in the [Phase 2 one-pager](https://docs.google.com/document/d/1X2TM_kZLZ-LZP4KdyNITQTdd9Ts1J2DQuNwoAEyT-3s/edit?usp=sharing&resourcekey=0-3n4FbMw2uSMKVEkOmbc_jQ) and the [migration guide](https://developers.google.com/analytics/devguides/migration/api/reporting-ua-to-ga4-dims-mets), update the mock metrics/dimensions as appropriate, and to remove those which are no longer relevant (i.e. the AdSense ones as GA4 isn't yet linked with AdSense).

**Comment by hussain-t** [comment]
Thanks for pointing it out, @techanvil. I have updated the IB.
